src/normalization.jl

```diff
@@ -0,0 +1,141 @@
+import OnlineStats: OnlineStats, Group, Moments, fit!, OnlineStat, Weight, EqualWeight, mean, std
+export state_normalizer, reward_normalizer, NormalizedTrajectory, Normalizer
+using MacroTools
+
+"""
+    Normalizer(::OnlineStat)
+
+Wraps an OnlineStat to be used by a [`NormalizedTrajectory`](@ref).
+"""
+struct Normalizer{OS<:OnlineStat}
+    os::OS
+end
+
+MacroTools.@forward Normalizer.os OnlineStats.mean, OnlineStats.std, Base.iterate, normalize!, Base.length
+
+function OnlineStats.fit!(n::Normalizer, y)
+    for yi in y
+        fit!(n, yi)
+    end
+    n
+end
+
+function OnlineStats.fit!(n::Normalizer, data::AbstractArray)
+    fit!(n.os, vec(data))
+    n
+end
+
+function OnlineStats.fit!(n::Normalizer, data::Number)
+    fit!(n.os, data)
+    n
+end
+
+"""
+    reward_normalizer(;weights = OnlineStats.EqualWeight())
+
+Returns preconfigured normalizer for scalar rewards. By default, all rewards have equal weights.
+See the [OnlineStats documentation](https://joshday.github.io/OnlineStats.jl/stable/weights/) to use variants such as exponential weights to favor the most recent observations.
+"""
+reward_normalizer(; weight::Weight = EqualWeight()) = Normalizer(Moments(weight = weight))
+
+"""
+    state_normalizer([state_size::Tuple{Int}]; weights = OnlineStats.EqualWeight())
+
+Returns preconfigured normalizer for scalar or array states. 
+For Array states, state_size is a tuple containing the dimension sizes of a state. E.g. `(10,)` for a 10-elements vector, or `(252,252)` for a square image.
+For scalar states, do not provide a state_size information.
+By default, all states have equal weights.
+See the [OnlineStats documentation](https://joshday.github.io/OnlineStats.jl/stable/weights/) to use variants such as exponential weights to favor the most recent observations.
+"""
+state_normalizer(; weight::Weight = EqualWeight()) = Normalizer(Moments(weight = weight))
+
+state_normalizer(state_size; weight::Weight = EqualWeight()) = Normalizer(Group([Moments(weight = weight) for _ in 1:prod(state_size)]))
+
+
+"""
+    NormalizedTrajectory(trajectory, normalizer::Dict{Symbol, Normalizer})
+    NormalizedTrajectory(trajectory, normalizer::Pair{Symbol, Normalizer}...)
+
+Wraps a `Trajectory` and a [`Normalizer`](@ref). When pushing new elements of `:trace` to trajectory, a `NormalizedTrajectory` will first update `normalizer[:trace]`, an online estimate of the mean and variance of :trace.
```

Comment:
Yes that's actually a better idea. I'll do that.

Reply:
What do you think of this API : 
```
t = Trajectory(
    container=Traces(
        a=NormalizedTrace(Int[], reward_normalizer()) ,
        b=Bool[]
    ),
    sampler=BatchSampler(3),
    controler=InsertSampleRatioControler(0.25, 4)
)
```
It means that we have to modify the Traces constructor because we don't want to convert NormalizedTrace back to Trace.